### PR TITLE
Dedupe references in the workflows

### PR DESCRIPTION
**Pull request by david-caro**
- [ ] I have all the information that I need (if not, move to `RFC` and look for it).
- [ ] I linked the related issue(s) in the corresponding commit logs.
- [ ] I wrote [good commit log messages](https://github.com/torvalds/subsurface-for-dirk/blob/5f15ad5a86ada3c5e574041a5f9d85235322dabb/README#L92-L119).
- [ ] My code follows the code style of this project.
- [ ] I've added any new docs if API/utils methods were added.
- [ ] I have updated the existing documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: tests/unit/workflows/test_workflows_tasks_submission.py [hunk 1]
```diff
@@ -33,6 +33,7 @@
     create_ticket,
     filter_keywords,
     prepare_keywords,
+    remove_references,
```
Comment (michamos) [inline tests/unit/workflows/test_workflows_tasks_submission.py]: why?

Comment (david-caro) [reply tests/unit/workflows/test_workflows_tasks_submission.py]: oops... pushed also the revert xd, one sec

**Comment by david-caro** [comment]
I'll add also 1 more test for the submission reference, but for the pdf is more complicated...

File: inspirehep/modules/workflows/tasks/actions.py [hunk 2]
```diff
@@ -425,11 +426,13 @@ def refextract(obj, eng):
         if tmp_document:
             pdf_references = extract_references_from_pdf(tmp_document, source)
             pdf_references = [match_reference(ref) for ref in pdf_references]
+            pdf_references = dedupe_list(pdf_references)
 
     text = get_value(obj.extra_data, 'formdata.references')
     if text:
         text_references = extract_references_from_text(text, source)
         text_references = [match_reference(ref) for ref in text_references]
+        text_references = dedupe_list(text_references)
```
Comment (turtle321) [inline inspirehep/modules/workflows/tasks/actions.py]: This is not tested, if I read correctly.

File: inspirehep/modules/workflows/tasks/actions.py [hunk 3]
```diff
@@ -425,11 +426,13 @@ def refextract(obj, eng):
         if tmp_document:
             pdf_references = extract_references_from_pdf(tmp_document, source)
             pdf_references = [match_reference(ref) for ref in pdf_references]
+            pdf_references = dedupe_list(pdf_references)
```
Comment (turtle321) [inline inspirehep/modules/workflows/tasks/actions.py]: This is not tested. I know it is not easy, so I am fine with skipping this test for now, but please create a US to point to this comment. I'll do it myself actually.

File: inspirehep/modules/workflows/tasks/actions.py (hunk 2 above)
Comment (david-caro) [reply inspirehep/modules/workflows/tasks/actions.py]: not yet yes, I'm adding a test for it.

File: inspirehep/modules/workflows/tasks/actions.py (hunk 3 above)
Comment (turtle321) [reply inspirehep/modules/workflows/tasks/actions.py]: https://its.cern.ch/jira/browse/INSPIR-652

**Comment by david-caro** [comment]
Superseded by https://github.com/inspirehep/inspire-next/pull/3398

**Pull request closed without merging**